alacritty/src/display/mod.rs

```diff
@@ -1125,19 +1130,22 @@ impl Display {
         rects.extend(underline.rects(Flags::UNDERLINE, &metrics, &self.size_info));
 
         let ime_popup_point = match preedit.cursor_end_offset {
-            Some(cursor_end_offset) if cursor_end_offset != 0 => {
-                let is_wide = preedit.text[preedit.cursor_byte_offset.unwrap_or_default()..]
-                    .chars()
-                    .next()
-                    .map(|ch| ch.width() == Some(2))
-                    .unwrap_or_default();
+            Some(cursor_end_offset) => {
+                // Regular edit happens when the start == end, thus use Beam for it, however
+                // when multiple characters are changed at once use the `HollowBlock`.
```

Comment:
```suggestion
                // Use hollow block when multiple characters are changed at once.
```

I think this should be sufficient.